v4/internal/context.go

```diff
@@ -0,0 +1,20 @@
+// Copyright 2020 New Relic Corporation. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package internal
+
+type contextKeyType struct{}
+
+var (
+	// TransactionContextKey is the key used for newrelic.FromContext and
+	// newrelic.NewContext.
+	TransactionContextKey = contextKeyType(struct{}{})
+
+	// GinTransactionContextKey is used as the context key in
+	// nrgin.Middleware and nrgin.Transaction.  Unfortunately, Gin requires
+	// a string context key. We use two different context keys (and check
+	// both in nrgin.Transaction and newrelic.FromContext) rather than use a
+	// single string key because context.WithValue will fail golint if used
+	// with a string key.
+	GinTransactionContextKey = "newRelicTransaction"
```

Comment:
These two vars are in the internal package because they are also used by our gin instrumentation.